x/slashing/querier.go

```diff
@@ -34,3 +45,86 @@ func queryParams(ctx sdk.Context, cdc *codec.Codec, k Keeper) ([]byte, sdk.Error
 
 	return res, nil
 }
+
+// QuerySigningInfoParams defines the params for the following queries:
+// - 'custom/slashing/signingInfo'
+type QuerySigningInfoParams struct {
+	ConsAddress sdk.ConsAddress
+}
+
+func NewQuerySigningInfoParams(consAddr sdk.ConsAddress) QuerySigningInfoParams {
+	return QuerySigningInfoParams{consAddr}
+}
+
+func querySigningInfo(ctx sdk.Context, cdc *codec.Codec, req abci.RequestQuery, k Keeper) ([]byte, sdk.Error) {
+	var params QuerySigningInfoParams
+
+	err := cdc.UnmarshalJSON(req.Data, &params)
+	if err != nil {
+		return nil, sdk.ErrInternal(fmt.Sprintf("failed to parse params: %s", err))
+	}
+
+	signingInfo, found := k.getValidatorSigningInfo(ctx, params.ConsAddress)
+	if !found {
+		return nil, ErrNoSigningInfoFound(DefaultCodespace, params.ConsAddress)
```

Comment:
so there's a good practice in returning `OK 200` even if the response is empty, because the request was processed successfully. See https://stackoverflow.com/a/13367198

Reply:
Hmmm, ok so for collections, I guess this makes sense. But this is a single resource you're looking for. If I go with this approach and just return a `200 OK` with an empty `ValidatorSigningInfo{}` object, you'll get the following as a response:

```json
{
    "address": "",
    "index_offset": "0",
    "jailed_until": "0001-01-01T00:00:00Z",
    "missed_blocks_counter": "0",
    "start_height": "0",
    "tombstoned": false
}
```

This isn't what I'd want to see as an API consumer. Or am I wrong?